src/meta-srv/src/procedure/region_failover/activate_region.rs

```diff
@@ -162,19 +164,26 @@ impl ActivateRegion {
                     // would be in vain. Then why not just end the failover procedure? Because we
                     // currently lack the methods or any maintenance tools to manage the whole
                     // procedures things, it would be easier to let the procedure keep running.
-                    let reason = format!(
+                    let violated = format!(
                         "Region {failed_region:?} is not opened by Datanode {:?}, error: {error:?}",
                         self.candidate,
                     );
-                    RetryLaterSnafu { reason }.fail()
+
+                    UnexpectedSnafu { violated }
+                        .fail()
+                        .map_err(BoxedError::new)
+                        .context(RetryLaterSnafu)
```

Comment:
Why add an indirection layer of `UnexpectedSnafu` to throw the retry error?

Reply:
PTAL, I refactor this part.